api/v1/reviewers/recommendations/ReviewerRecommendationController.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+/**
+ * @file api/v1/reviewers/recommendations/ReviewerRecommendationController.php
+ *
+ * Copyright (c) 2025 Simon Fraser University
+ * Copyright (c) 2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ReviewerRecommendationController
+ *
+ * @brief API controller class to handle actions on reviewer recommendations
+ *
+ */
+
+namespace APP\API\v1\reviewers\recommendations;
+
+use APP\API\v1\reviewers\recommendations\formRequests\AddReviewerRecommendation;
+use APP\API\v1\reviewers\recommendations\formRequests\EditReviewerRecommendation;
+use APP\API\v1\reviewers\recommendations\formRequests\UpdateStatusReviewerRecommendation;
+use APP\API\v1\reviewers\recommendations\resources\ReviewerRecommendationResource;
+use APP\core\Application;
+use APP\security\authorization\RecommendationAccessPolicy;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Route;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\authorization\UserRolesRequiredPolicy;
+use PKP\security\Role;
+use PKP\submission\reviewer\recommendation\ReviewerRecommendation;
+
+class ReviewerRecommendationController extends PKPBaseController
+{
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getHandlerPath()
+     */
+    public function getHandlerPath(): string
+    {
+        return 'reviewers/recommendations';
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getRouteGroupMiddleware()
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
+            self::roleAuthorizer([
+                Role::ROLE_ID_SITE_ADMIN,
+                Role::ROLE_ID_MANAGER,
+                Role::ROLE_ID_SUB_EDITOR,
+            ]),
+        ];
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::authorize()
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $illuminateRequest = $args[0]; /** @var \Illuminate\Http\Request $illuminateRequest */
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+        if (in_array($actionName, ['edit', 'updateStatus', 'delete'])) {
+            $this->addPolicy(
+                new RecommendationAccessPolicy(
+                    $request,
```

Comment:
Same as above: `$request` here is a `PKPRequest` but `RecommendationAccessPolicy` is expecting the more specific `APP\core\Request`.

Reply:
same as https://github.com/pkp/ojs/pull/4505#discussion_r2041122202 .